### PR TITLE
Read program configs from environment variables #519

### PR DESCRIPTION
**Pull request by archibate**
Related issue id = #519

### CODE REVIEW

File: python/taichi/lang/__init__.py [hunk 1]
```diff
@@ -77,6 +77,14 @@ def init(default_fp=None, default_ip=None, print_preprocessed=None, debug=None,
 
   for k, v in kwargs.items():
     setattr(ti.cfg, k, v)
+
+  if os.environ.get("TI_PRINT_IR") is not None:
+    ti.print_ir = True
```
Comment (yuanming-hu) [inline python/taichi/lang/__init__.py]: Should this be `ti.cfg.print_ir=True`?

File: python/taichi/lang/__init__.py [hunk 2]
```diff
@@ -77,6 +77,14 @@ def init(default_fp=None, default_ip=None, print_preprocessed=None, debug=None,
 
   for k, v in kwargs.items():
     setattr(ti.cfg, k, v)
+
+  if os.environ.get("TI_PRINT_IR") is not None:
```
Comment (yuanming-hu) [inline python/taichi/lang/__init__.py]: What if the user does `export TI_PRINT_IR=0`?

File: python/taichi/lang/__init__.py [hunk 3]
```diff
@@ -77,6 +77,14 @@ def init(default_fp=None, default_ip=None, print_preprocessed=None, debug=None,
...
+    ti.print_ir = True
+  arch = os.environ.get("TI_ARCH")
+  if arch is not None:
+    ti.cfg.arch = getattr(core.Arch, arch)
```
Comment (yuanming-hu) [inline python/taichi/lang/__init__.py]: `arch` is a string and should be translated to e.g. `ti.cuda`.

Comment (archibate) [reply python/taichi/lang/__init__.py]: You mean, use a `dict()`?

File: python/taichi/lang/__init__.py (hunk 2 above)
Comment (archibate) [reply python/taichi/lang/__init__.py]: if user want `print_ir = False`, they're expected to `unset TI_PRINT_IR`.

File: python/taichi/lang/__init__.py (hunk 3 above)
Comment (yuanming-hu) [reply python/taichi/lang/__init__.py]: Added `arch_from_name`: 96b116ee

**Pull request merged**